### PR TITLE
MULE-10160 - Move lifecycle interfaces to mule-api

### PR DESCRIPTION
**Pull request by pablolagreca**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/lifecycle/Startable.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+public interface Startable {
+
+  String PHASE_NAME = "start";
+
+  void start() throws MuleException;
+
+  /**
+   * Determines if this object is started or not
+   */
+  // TODO MULE-3969
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/lifecycle/Startable.java]: remove

File: src/main/java/org/mule/runtime/api/lifecycle/StartException.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.api.lifecycle;
+
+import org.mule.runtime.api.i18n.I18nMessage;
+
+/** <code>DisposeException</code> TODO (document class) */
+
+public class StartException extends LifecycleException {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/lifecycle/StartException.java]: I'm pretty sure we almost never use these Start/Stop/Bleh/Exceptions. We just go with MuleException or InitialisationException. is it worth cleaning up?

File: src/main/java/org/mule/runtime/api/util/ComponentLocationProvider.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import org.mule.runtime.api.meta.AnnotatedObject;
+
+import javax.xml.namespace.QName;
+
+/**
+ * Provides a standard way to generate a log entry or message that references an element in a flow.
+ *
+ * @since 4.0
+ */
+public class ComponentLocationProvider {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/util/ComponentLocationProvider.java]: having this in mule-api feels weird.... maybe it's fine... but weird

File: src/main/java/org/mule/runtime/api/lifecycle/StartException.java (hunk 2 above)
Comment (pablolagreca) [inline src/main/java/org/mule/runtime/api/lifecycle/StartException.java]: done

File: src/main/java/org/mule/runtime/api/exception/LocationResolver.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.api.exception;
+
+public class LocationResolver {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/exception/LocationResolver.java]: genetic garbage?

Comment (pablolagreca) [inline src/main/java/org/mule/runtime/api/exception/LocationResolver.java]: my chromosome count is getting lower than usual.

**Pull request merged**